src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -390,13 +611,12 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
             //
             // If none of those are the case then we can just take an early out.
 
-            ArrayBuilder<LocalSymbol> temporariesBuilder = ArrayBuilder<LocalSymbol>.GetInstance();
-            rewrittenArguments = _factory.MakeTempsForDiscardArguments(rewrittenArguments, temporariesBuilder);
+            temps ??= ArrayBuilder<LocalSymbol>.GetInstance();
+            rewrittenArguments = _factory.MakeTempsForDiscardArguments(rewrittenArguments, temps);
```

Comment:
>_factory.MakeTempsForDiscardArguments(rewrittenArguments, temps); [](http://example.com/codeflow?start=33&length=65)

Should this be done during loering of arguments? Presumable we need this argument to create a builder?

Reply:
Yeah, you're right. Unlikely scenario, certainly, but we should do this earlier.